pyomo/core/tests/unit/test_derivs.py

```diff
@@ -300,3 +305,78 @@ def test_nested_named_expressions(self):
         self.assertAlmostEqual(derivs[m.x], approx_deriv(e, m.x), tol)
         self.assertAlmostEqual(derivs[m.y], pyo.value(symbolic[m.y]), tol+3)
         self.assertAlmostEqual(derivs[m.y], approx_deriv(e, m.y), tol)
+
+class TestDifferentiate(unittest.TestCase):
+    @unittest.skipUnless(sympy_available, "test requires sympy")
+    def test_sympy(self):
+        m = pyo.ConcreteModel()
+        m.x = pyo.Var(initialize=0.23)
+        m.y = pyo.Var(initialize=0.88)
+        ddx = differentiate(m.x**2, wrt=m.x, mode='sympy')
+        self.assertIsInstance(ddx, MonomialTermExpression)
+        self.assertEqual(str(ddx), '2.0*x')
+        self.assertAlmostEqual(ddx(), 0.46)
+        ddy = differentiate(m.x**2, wrt=m.y, mode='sympy')
+        self.assertEqual(ddy, 0)
+
+        ddx = differentiate(m.x**2, wrt_list=[m.x, m.y], mode='sympy')
+        self.assertIsInstance(ddx, list)
+        self.assertEqual(len(ddx), 2)
+        self.assertIsInstance(ddx[0], MonomialTermExpression)
```

Comment:
This is a good place for the `compare_expressions` funtion (https://github.com/Pyomo/pyomo/blob/2ad064be83da890f07fb1cd6109aca6ed4916c69/pyomo/core/expr/compare.py#L159)

Reply:
Done.